src/app/modules/main/chat_section/module.nim

```diff
@@ -690,6 +692,16 @@ proc addNewChat(
       else:
         categoryPosition = category.position
 
+  var canPostReactions = true
+  var hideIfPermissionsNotMet = false
+  var viewersCanPostReactions = true
+  if self.controller.isCommunity:
+    let communityChat = community.getSpecificCommunityChat(chatDto.id)
```

Comment:
Not sure I understand this part, isn't `communityChat` the same type (`ChatDto`) as `chatDto` that is already processed? Aren't we basically refetching the same `chatDto` here?

Reply:
Same thing as explained above. The CommunityChat contains more info still. It will be addressed in a further refactor.